### ISSUE TITLE
[BUG] Preview Windows fail to hide when not hovered

### ISSUE THREAD

**Issue by RaimundasSakalauskas** [ejbills/DockDoor#396]
## Describe the bug
Preview window fails to hide if mouse never hovers other app or preview window.

## Steps to reproduce

Bug is very easy to repro with dock autohide:
1) Hover any app to see previews
2) Move the mouse to the gap between dock and preview window
3) When auto dock hides, move the mouse anywhere avoiding hovering the preview view

The view will remain visible on screen until hovered.

I am using dock on the left side of the screen.

## Expected behavior
Hide preview on whichever happens first:
1) Another app icon is hovered in the dock
2) If mouse is moved away from projected area:
<img width='400' src="https://github.com/user-attachments/assets/a83df758-cb7d-4c34-b1b6-95fd11f43ee1">

## Actual behavior
The preview window remained visible until hovered.

## Environment
 - macOS version: 15.1.1
 - App version: 1.3.3

- [x] I have reviewed the open issues to ensure this is not a duplicate

**Comment by ejbills** [comment]
I am closing this issue as there is a number of remedies:

- If the mouse moves significantly away from an active hover preview, the preview will close itself automatically. This is something like 800px. 
- You can reduce the buffer of the window preview so that it aligns with the dock without any gap. 

If you are experiencing this with regular usage, let me know. Thanks!

**Comment by RaimundasSakalauskas** [comment]
That is the reason I have filed this - since I use auto-hide, over the last couple of days I have experienced this a dozen of times. Knowing about 800px helps a little, but on 2560 rendering resolution, it means that it only disappears if mouse crosses the middle of the screen (give or take).

Overall great product, that has 4 problems that I have noticed and too lazy to look into fixing myself (at least for now):
1) The problem described in the ticket
2) Not sure how hypedock did that, but they were able to keep dock from hiding without using macos prefs. Using prefs is easy, but apps like Xcode resize the window when the flag toggles. I'd expect that they use private APIs.
3) Previews and settings window fails to render over anything else. Previews render underneath dock labels and to see settings I have to use expose as it hides underneath other windows when its opened.
4) Previews don't hide when mouse moves to a "neighbor" app that has no active windows. This would partially be mitigated with the proposed solution.
